linera-core/src/value_cache.rs

```diff
@@ -151,18 +151,26 @@ impl CertificateValueCache {
     /// [`HashedCertificateValue::ConfirmedBlock`] is also cached.
     pub async fn insert<'a>(&self, value: Cow<'a, HashedCertificateValue>) -> bool {
         let hash = value.hash();
+        let maybe_confirmed_value = value.validated_to_confirmed();
         let mut cache = self.cache.lock().await;
         if cache.contains(&hash) {
-            return false;
-        }
-        if let Some(confirmed_value) = value.validated_to_confirmed() {
-            // Cache the certificate for the confirmed block in advance, so that the clients don't
-            // have to send it.
-            cache.push(confirmed_value.hash(), confirmed_value);
+            // Promote the re-inserted value in the cache, as if it was accessed again.
+            cache.promote(&hash);
```

Comment:
(Promoting one value can never cause eviction of another, can it? I assume the cache simply has a maximum capacity, and things can only get evicted when others are inserted?)

Reply:
That's correct 👍